bottlerocket-settings-models/settings-extensions/container-runtime-plugins/src/de.rs

```diff
@@ -0,0 +1,209 @@
+use byte_unit::Byte;
+use serde::{self, Deserialize, Deserializer};
+
+/// This function deserializes a size string (e.g., "64mb") or integer into an i64 value.
+///
+/// Valid formats:
+/// - Integer values (e.g., "1024" or 1024)
+/// - Size strings with units (e.g., "64mb", "1.5gb", ".5gb")
+///
+/// Invalid formats:
+/// - Negative values other than -1
+/// - Strings with invalid formats
+/// - Strings with whitespace
+pub fn deserialize_chunk_size<'de, D>(deserializer: D) -> Result<Option<i64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum StringOrIntOrNone {
+        String(String),
+        Int(i64),
+        None,
```

Comment:
I don't think we want to support explicit nulls in settings. The semantics around that would be a bit murky, like the value would be set, but to an "unset" value. Seems likely to trip something up.

Reply:
In this same vein, given that we're taking some liberty with the data input, I wonder if it would be a good idea to support "unlimited" as an alternative to "-1".

Reply:
Latest force push removes the explicit `None`.